lib/managers/folders.js

```diff
@@ -301,6 +301,45 @@ Folders.prototype.deleteMetadata = function(folderID, scope, template, callback)
 	this.client.del(apiPath, null, this.client.defaultResponseHandler(callback));
 };
 
+/**
+ * Restores an item that has been moved to the trash. Default behavior is to restore the item
+ * to the folder it was in before it was moved to the trash. If that parent folder no longer
+ * exists or if there is now an item with the same name in that parent folder, the new parent
+ * older and/or new name will need to be included in the request.
+ *
+ * API Endpoint: '/folders/:folderID'
+ * Method: POST
+ *
+ * @param {string} folderID - The ID of the folder to restore
+ * @param {?string} [name] - The new name for this item
+ * @param {string} [parentFolderID] - The new parent folder for this item
+ * @param {Function} callback - Called with folder information if successful, error otherwise
+ * @returns {void}
+ */
+Folders.prototype.restoreFolder = function(folderID, name, parentFolderID, callback) {
+
+	var params = {
+		body: {}
+	};
+
+	if (name) {
+		Object.assign(params.body, {
+			name: name
+		});
+	}
+
+	// id can be 0
+	if (parentFolderID != null) {
+		Object.assign(params.body, {
```

Comment:
Sure , Object.assign is just personal preference and I saw it being used elsewhere but that was for full fileData object.